eppoclient/rules.go

```diff
@@ -50,7 +50,13 @@ func (condition condition) matches(subjectAttributes Attributes) bool {
 		subjectValueNumeric, isNumericSubjectErr := toFloat64(subjectValue)
 		conditionValueNumeric, isNumericConditionErr := toFloat64(condition.Value)
 		if isNumericSubjectErr == nil && isNumericConditionErr == nil {
-			return evaluateNumericCondition(subjectValueNumeric, conditionValueNumeric, condition)
+			result, err := evaluateNumericCondition(subjectValueNumeric, conditionValueNumeric, condition)
+			if err != nil {
+				// Decision: Should we swallow the error here and return false?
+				// Alternatively, we could log the error and return false.
+				return false
```

Comment:
@greghuels What is most helpful to add foundationally here to support evaluation reasons? I am thinking changing the function definition to `(bool, error)` so we can give the highest fidelity error up to the `getAssignment` method about the nature of the failure.

Reply:
See my other comment: `evaluateNumericCondition` has no legitimate reason to return an error. An error here means we have a bug in SDK.

Reply:
+1 to changing the signature here to return `(bool, error)` - we don't want to just silently swallow it

Reply:
I do think we should surface _unexpected_ errors the SDK encounters. So I side with the above two comments.

Anticipated error conditions, like the user passing a string-valued attribute for comparison using a numeric operator, should return `false` like we do.

Side note: it looks like in JavaScript, if the attribute isn't a number, we just compare the non-number as a string to the target number as a string 😬 I can file a bug for that.

Reply:
There are a couple of other options to achieve this:
- Inline these functions in the call site. The call site has the proper guards, so error branch becomes unreachable—there will be no panics nor errors.
- Wrap the whole evaluation into a recover, so the panic does not reach the user. It's probably a good idea to do this anyway—there are many more ways Go can panic without an explicit `panic` call

Reply:
> Wrap the whole evaluation into a recover, so the panic does not reach the user. It's probably a good idea to do this anyway because we can't guard against all potential panics

Yea this makes sense to me - will catch the panic and return default value. Great suggestion.